packages/fxa-settings/src/components/App/index.test.tsx

```diff
@@ -22,42 +28,58 @@ const appProps = {
   config: { metrics: { navTiming: { enabled: false, endpoint: '' } } },
 };
 
-beforeEach(() => {
-  //@ts-ignore
-  delete window.location;
-  window.location = {
-    ...window.location,
-    replace: jest.fn(),
-  };
-});
+describe('App', () => {
+  beforeEach(() => {
+    //@ts-ignore
+    delete window.location;
+    window.location = {
+      ...window.location,
+      replace: jest.fn(),
+    };
+  });
 
-it('renders', async () => {
-  await act(async () => {
-    render(<App {...appProps} />);
+  it('renders', async () => {
+    await act(async () => {
+      render(<App {...appProps} />);
+    });
   });
-});
 
-it('Initializes metrics flow data when present', async () => {
-  const DEVICE_ID = 'yoyo';
-  const BEGIN_TIME = 123456;
-  const FLOW_ID = 'abc123';
-  const flowInit = jest.spyOn(Metrics, 'init');
-  const updatedAppProps = Object.assign(appProps, {
-    flowQueryParams: {
+  it('Initializes metrics flow data when present', async () => {
+    const DEVICE_ID = 'yoyo';
+    const BEGIN_TIME = 123456;
+    const FLOW_ID = 'abc123';
+    const flowInit = jest.spyOn(Metrics, 'init');
+    const updatedAppProps = Object.assign(appProps, {
+      flowQueryParams: {
+        deviceId: DEVICE_ID,
+        flowBeginTime: BEGIN_TIME,
+        flowId: FLOW_ID,
+      },
+    });
+
+    await act(async () => {
+      render(<App {...updatedAppProps} />);
+    });
+
+    expect(flowInit).toHaveBeenCalledWith(true, {
       deviceId: DEVICE_ID,
-      flowBeginTime: BEGIN_TIME,
       flowId: FLOW_ID,
-    },
+      flowBeginTime: BEGIN_TIME,
+    });
+    expect(window.location.replace).not.toHaveBeenCalled();
   });
 
-  await act(async () => {
-    render(<App {...updatedAppProps} />);
-  });
+  it('redirects `settings/avatar/change` to `settings/avatar`', async () => {
+    renderWithRouter(
+      <AppContext.Provider value={mockAppContext({ account })}>
+        <App {...appProps} />
+      </AppContext.Provider>,
+      {
+        route: `/settings/avatar/change`,
+      }
+    );
 
-  expect(flowInit).toHaveBeenCalledWith(true, {
-    deviceId: DEVICE_ID,
-    flowId: FLOW_ID,
-    flowBeginTime: BEGIN_TIME,
+    screen.debug();
+    expect(screen.getByTestId('flow-container')).toBeInTheDocument();
```

Comment:
I think we have this on every page. I know the test isn't passing yet but maybe test for some avatar-related element here similar to the functional test.

I'm also thinking we probably don't need both a functional test and a unit test for this but you've already got most of the work done so keeping both is OK. I like unit tests for the line coverage reports and functional tests to be more like end to end testing for some flow that a user is going through but I know we already had another redirect test in the functional tests.